src/gradualizer_lib.erl

```diff
@@ -170,3 +177,32 @@ get_ast_children({clauses, Clauses}) ->
 get_ast_children(Node) ->
     [_Tag, _Anno | Children] = tuple_to_list(Node),
     Children.
+
+
+%% ------------------------------------------------
+%% Type environment
+%% ------------------------------------------------
+
+-spec empty_tenv() -> tenv().
+empty_tenv() ->
+    #{module => undefined,
+      types => #{},
+      records => #{}}.
+
+-spec create_tenv(_, _, _) -> tenv().
+create_tenv(Module, TypeDefs, RecordDefs) ->
```

Comment:
Specs like this one can always be improved or be accompanied by some comment. I know it's just moved code so I won't request it as part of this PR.